android/build.gradle

```
@@ -368,16 +366,56 @@ task createNativeDepsDirectories(dependsOn: applyJavaPatches) {
     thirdPartyNdkDir.mkdirs()
 }
 
-task downloadBoost(dependsOn: createNativeDepsDirectories, type: Download) {
+def resolveTaskFactory(String name, String artifactLocalName, File reactNativeAndroidDownloadDir, File reanimatedDownloadDir) {
+    return tasks.create(name: name.startsWith("resolve") ? name : "resolve" + name, dependsOn: createNativeDepsDirectories, type: Copy) {
+        from reactNativeAndroidDownloadDir
+        include artifactLocalName
+        into reanimatedDownloadDir
+
+        onlyIf {
+            // First we check whether the file is already in our download directory
+            if (file("$reanimatedDownloadDir/$artifactLocalName").isFile()) {
+                return false
+            }
+
+            // If it is not the case we check whether it was downloaded by ReactAndroid project
+            def tarFile = Paths.get(reactNativeAndroidDownloadDir.toString(), artifactLocalName).toFile()
+            return tarFile.isFile()
```

1. Let's be more explicit with the logic here.
2. Let's unify the code in both cases.
```suggestion
            if (file("$reactNativeAndroidDownloadDir/$artifactLocalName").isFile()) {
                return true
            }
            
            return false
```

---

[56df771](https://github.com/software-mansion/react-native-reanimated/pull/3588/commits/56df7718fc50633b23bc91dc6d4ce8821990a487)

---

I actually thought using `file(...)` won't work, because it is method defined on gradle's `Project` type & it resolves files relatively to current project directory, but I guess the case is we are passing absolute path there so it actually works (tested) 🤷🏻‍♂️